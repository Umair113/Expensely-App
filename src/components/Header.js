import React from 'react';
import { NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = ({startLogout}) => (
    <header className="header">
        <div className ="header-container">
            <div className="content-container">
                <h1>Expense-Ly</h1>
            </div>
        </div>

        <div className='nav-toolbox-separator'>
            <div className="content-container">
                <div className="nav-container">
                    <div className="nav-links">
                        <NavLink to="/dashboard" className = 'nav-item' activeClassName="is-active" >Dashboard</NavLink>
                        <NavLink to="/help" className = 'nav-item' activeClassName="is-active" >Help</NavLink>
                    </div>
                    <div className = "nav-logout">
                        <button className='nav-item' onClick={startLogout}>Logout</button>
                    </div>
                </div>
            </div>
        </div>

    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
