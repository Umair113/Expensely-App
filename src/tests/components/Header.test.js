// react-test-renderer

import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('should render Head correctly', ()=>{
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
})