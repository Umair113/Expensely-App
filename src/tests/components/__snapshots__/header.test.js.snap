// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="header-container"
  >
    <div
      className="content-container"
    >
      <h1>
        Expense-Ly
      </h1>
    </div>
  </div>
  <div
    className="nav-toolbox-separator"
  >
    <div
      className="content-container"
    >
      <div
        className="nav-container"
      >
        <div
          className="nav-links"
        >
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            className="nav-item"
            to="/dashboard"
          >
            Dashboard
          </NavLink>
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            className="nav-item"
            to="/help"
          >
            Help
          </NavLink>
        </div>
        <div
          className="nav-logout"
        >
          <button
            className="nav-item"
            onClick={[Function]}
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </div>
</header>
`;
